autoscale/src/main/java/com/sequenceiq/periscope/monitor/handler/CloudbreakCommunicator.java

```diff
@@ -33,20 +42,27 @@
     @Inject
     private RequestLogging requestLogging;
 
+    @Retryable(value = Exception.class, maxAttempts = 5, backoff = @Backoff(delay = 10000))
     public StackV4Response getByCrn(String stackCrn) {
         return cloudbreakInternalCrnClient.withInternalCrn().autoscaleEndpoint().get(stackCrn);
     }
 
+    @Retryable(value = Exception.class, maxAttempts = 5, backoff = @Backoff(delay = 10000))
     public AutoscaleStackV4Response getAutoscaleClusterByCrn(String stackCrn) {
         return cloudbreakInternalCrnClient.withInternalCrn()
                 .autoscaleEndpoint().getAutoscaleClusterByCrn(stackCrn);
     }
 
+    @Retryable(value = Exception.class, maxAttempts = 5, backoff = @Backoff(delay = 10000))
     public AutoscaleStackV4Response getAutoscaleClusterByName(String stackName, String accountId) {
         return cloudbreakInternalCrnClient.withInternalCrn()
                 .autoscaleEndpoint().getInternalAutoscaleClusterByName(stackName, accountId);
     }
 
+    public AutoscaleStackV4Responses getAutoscaleClustersByCrns(List<String> crns) {
+        return new AutoscaleStackV4Responses(crns.stream().map(this::getAutoscaleClusterByCrn).collect(toList()));
```

Comment:
the solution here is the same: API call in foreach, so in the end, this would result the same: multiple API calls towards CB

Reply:
Implemnted it this way because I didnt find a listByCrn equivalent in CB core. Would it make sense to introduce a new endpoint for this and use that instead?

Reply:
yes, that would be useful and it would be probably not much effort to introduce a list API in AutoscaleV4Controller in CB core.